stencil_main.py

```diff
@@ -222,299 +176,163 @@ def main(
-        if stencil_name == "test":
-            stencils_numpy.test(in_field)
-
-        if stencil_name == "laplacian1d":
-            stencils_numpy.laplacian1d(in_field, tmp_field, num_halo=num_halo, extend=0)
-
-        if stencil_name == "laplacian2d":
-            stencils_numpy.laplacian2d(in_field, tmp_field, num_halo=num_halo, extend=0)
-
-        if stencil_name == "laplacian3d":
-            stencils_numpy.laplacian3d(in_field, tmp_field, num_halo=num_halo, extend=0)
-            
-        if stencil_name == "FMA":
-            stencils_numpy.FMA(in_field, in_field2, in_field3, tmp_field, num_halo=num_halo, extend=0)
-        
-        if stencil_name == "lapoflap1d":
-            stencils_numpy.lapoflap1d(in_field, tmp_field, tmp_field, num_halo=2, extend=1)
-
-        if stencil_name == "lapoflap2d":
-            stencils_numpy.lapoflap2d(in_field, tmp_field, tmp_field, num_halo=2, extend=1)
-
-        if stencil_name == "lapoflap3d":
-            stencils_numpy.lapoflap3d(in_field, tmp_field, tmp_field, num_halo=2, extend=1)
-    
-    if backend == "numbajit":
-        #if stencil_name == "laplacian1d":
-        #    stencil = njit(stencils_numpy.laplacian1d, parallel=True, cache=True, fastmath=False)
-        #    stencil(in_field, tmp_field, num_halo=num_halo, extend=0)
-            
-        if stencil_name == "test":
-            stencils_numbajit.test(in_field)
-
-        if stencil_name == "laplacian1d":
-            stencils_numbajit.laplacian1d(in_field, tmp_field, num_halo=num_halo, extend=0)
-            #print(stencils_numbajit.laplacian1d.inspect_types())
-
-        if stencil_name == "laplacian2d":
-            stencils_numbajit.laplacian2d(in_field, tmp_field, num_halo=num_halo, extend=0)
-
-        if stencil_name == "laplacian3d":
-            stencils_numbajit.laplacian3d(in_field, tmp_field, num_halo=num_halo, extend=0)
-            
-        if stencil_name == "FMA":
-            stencils_numbajit.FMA(in_field, in_field2, in_field3, tmp_field, num_halo=num_halo, extend=0)
-        
-#         if stencil_name == "lapoflap1d":
-#             stencils_numbajit.lapoflap1d(in_field, tmp_field, tmp_field, num_halo=2, extend=1)
+    # convert numpy arrays into gt4py storages
+    if backend == "gt4py":
+        in_field = gt4py.storage.from_array(
+            in_field, gt4py_backend, default_origin=(num_halo, num_halo, num_halo)
+        )
+        in_field2 = gt4py.storage.from_array(
+            in_field2, gt4py_backend, default_origin=(num_halo, num_halo, num_halo)
+        )
+        in_field3 = gt4py.storage.from_array(
+            in_field3, gt4py_backend, default_origin=(num_halo, num_halo, num_halo)
+        )
+        tmp_field = gt4py.storage.from_array(
+            tmp_field, gt4py_backend, default_origin=(num_halo, num_halo, num_halo)
+        )
+        out_field = gt4py.storage.from_array(
+            out_field, gt4py_backend, default_origin=(num_halo, num_halo, num_halo)
+        )

```

Comment:
GT4Py stencils cannot be fed with plain Numpy arrays, but with dedicated storages (which can be  created starting from Numpy arrays).